# Analysis of Data

## *** Questions to be addressed for analysing the dataset. *** ##

In [25]:
import numpy as np
import pandas as pd
import os

import seaborn as sns
import matplotlib.pyplot as plt
sns.set(style="white")
path = "C:/Users/viki4/Desktop/Data Science/Oil_Prices.csv"
data = pd.read_csv(path, parse_dates=True)
data.head(10)

Date   Open   High    Low   Last  Change  Settle  Volume  \
0  10/10/2017  49.55  51.06  49.54  50.94    1.34   50.92  641402   
1   10/9/2017  49.25  49.79  49.13  49.56    0.29   49.58  487932   
2   10/6/2017  50.75  50.82  49.10  49.25    1.50   49.29  720752   
3   10/5/2017  49.88  51.22  49.85  50.75    0.81   50.79  649264   
4   10/4/2017  50.16  50.67  49.76  49.90    0.44   49.98  596897   
5   10/3/2017  50.59  50.73  50.14  50.18    0.16   50.42  460593   
6   10/2/2017  51.64  51.71  50.07  50.57    1.09   50.58  597671   
7   9/29/2017  51.58  51.77  51.25  51.64    0.11   51.67  550344   
8   9/28/2017  52.05  52.86  51.22  51.59    0.58   51.56  799316   
9   9/27/2017  52.09  52.34  51.63  52.03    0.26   52.14  631972   

   Previous Day Open Interest  
0                    407051.0  
1                    453631.0  
2                    495395.0  
3                    511107.0  
4                    513316.0  
5                    518158.0  
6                    522982.0  
7                    529373.0  
8                    540544.0  
9                    546765.0

In [28]:
data.head(50).describe()

Open       High        Low       Last     Change     Settle  \
count  50.000000  50.000000  50.000000  50.000000  49.000000  50.000000   
mean   49.076400  49.636400  48.448200  49.081800   0.633469  49.088600   
std     1.543784   1.501472   1.552813   1.551044   0.475572   1.558779   
min    45.950000  46.720000  45.580000  45.970000   0.020000  45.960000   
25%    47.762500  48.455000  47.162500  47.747500   0.230000  47.697500   
50%    49.115000  49.685000  48.530000  49.130000   0.500000  49.165000   
75%    49.902500  50.665000  49.507500  49.922500   1.010000  49.962500   
max    52.120000  52.860000  51.630000  52.130000   1.610000  52.220000   

             Volume  Previous Day Open Interest  
count  5.000000e+01                   50.000000  
mean   6.623163e+05               412070.400000  
std    2.472855e+05               171306.102072  
min    2.521500e+04                25636.000000  
25%    5.962940e+05               305781.250000  
50%    7.035355e+05               512211.500000  
75%    8.243565e+05               528943.750000  
max    1.113414e+06               592595.000000

## *** Analysis of data that deals with the prices of oil in a particular month. *** ##

<li>Here, we consider the data that deals with the Opening, Closing, Highest and lowest price of oil in a particular month.  
<li> Also, the data considers the price that is settled for the particular day, i.e the final price of oil for that particular day.
<li> The data also specifies the volume of oil sold.

*** Based on the above data, the following questions can be asked about the dataset. ***

In [16]:
## Question 1.

# 1. What can be the total expenditure on oil for the entire year ? #

# 2.  What data will be used to answer the question?  #
<li> In order to calculate the total expenditure on oil, we con consider the columns "Settle", which means the final price of oil on the particular day and "Volume" which denotes the total volume of oil sold on that day.

# 3. Describe a second way to answer the same question? #
<li> The alternate way to address the same problem could be Exploratory Data Analysis.
<li> Here, we can analyze the weekly or monthly percentage increase/decrease in the price, by plotting a graph of percent increase/decrease versus months. 
<li> Using this data, we can predict the expenditure in a month and eventually, of an year.

# 4. What are assumptions and limitations of the approach? #
<li> To calculate the total expenditure on the oil, we have assumed that "Settle" is the final price of oil for the particular day. 
<li> For example, say, for the first day, i.e. on 10/10/2017, the Settled price of oil is 50.92 and the volume sold is 641402.
<li> Based on these values, we can predict the expenditure on oil on 10/10/2017 => 641402 x 50.92 = 32660190
<li> On the second day, i.e. on 10/9/2017 => 487932 x 49.58 = 24191669
<li> So, the expenditure for the total week could be the average of the total expenditure for 7 days, say ~26000000.
<li> Now, based on the average for 7 days, we can predict the expenditure for an year, i.e. ~26000000 x 4 x 12 = ~1248000000.
<li> The limitations of this appraoch is that, the price of oil considered for calculating the expenditure may not be the actual value, due to which, the calculated price may not be accurate. 
<li> Also, there is a possibility of the price of oil as well as currency falling gradually in the following year.
<li> The expenditure may also be inclusive of taxes.
<li> This value is estimated assuming that the oil is sold on all the 365 days of the year, which is also a limitation in itself as the oil may not be sold everyday.

# 5. What is the statement that you will be able to make after doing the analysis ? #
<li> The expenditure on oil may not be solely dependent on the settled price and the volume of oil. 

In [15]:
## Question 2.

# 1.  What could be the price of oil in the future ? #


# 2. What data will be used to answer the question? #
<li> Here, we can predict the price of oil by considering the mean of columns Open, High, Low and Last.
<li> For example, the opening price of oil in the first week is ~50, High is ~50, Low is ~49 and Last is ~49. (Average of the "Open", "High", "Low" and "Last" values for a week)
<li> Now, considering the mean of all the prices, we can predict the price of oil in the future to be ~49.50.

# 3. Describe a second way to answer the same question? # 
<li> The alternate way to estimate the price could be by plotting the graph of High vresus Low prices using which, we can predict the limits within which the prices would rise or fall for a particular year. 
<li> We can plot a graph to analyze the pattern of rise or fall in the prices.
<li> Also, using data.describe() as shown above, we can predict the increase or decrease in price by analysing the mean value for the recent entries in the dataset, which turns out to be 49.07.


# 4.  What are assumptions and limitations of the approach? #
<li> The price of oil is assumed to be the mean values of all the prices from the history of the previous year.
<li> The data consists of nul values, which are considered to be the mean of the the previous and next day. 
<li> However, it may be possible that the price may rise or fall to a whole new level in the future.
<li> Also, the prices mentioned in the data may be inconsistent.

# 5. What is the statement that you will be able to make after doing the analysis ? #
<li> The price of oil cannot be estimated just by calculating the mean of the prices from the data.
<li> The prices may turn out to be a completely new set of figures in the future. 